build: add workflow to generate monthly changelog (#5983)
stdlib-js/stdlib: lib/node_modules/@stdlib/_tools/changelog/generate/README.md

-   **content**: Markdown formatted changelog.
-   **releaseType**: release type (`null` if changelog is for a non-release).

The function accepts the following `options`:

> -   **releaseType**: a release type for which to generate the changelog for.
> -   **flags**: `git log` options used to retrieve commits to generate the changelog for.

Reviewer: ```suggestion
-   **releaseType**: a release type for which to generate the changelog.
-   **flags**: `git log` options used to retrieve commits from which to generate the changelog.
```

-   **content**: Markdown formatted changelog.
-   **releaseType**: release type (`null` if changelog is for a non-release).

The function accepts the following `options`:

-   **releaseType**: a release type for which to generate the changelog.

> -   **format**: whether to generate a `grouped` changelog where commits are grouped by package or `aggregated` one.

Reviewer: ```suggestion
-   **format**: changelog format. Must be one of the following:

    -   `'grouped'`: group commits by package.
    -   `'aggregated'`: TODO:add description.
```

Given the sub-list, you'll need to insert blank lines to satisfy Markdown linting.

-   **content**: Markdown formatted changelog.
-   **releaseType**: release type (`null` if changelog is for a non-release).

The function accepts the following `options`:

-   **releaseType**: a release type for which to generate the changelog.

-   **format**: changelog format. Must be one of the following:

    -   `'grouped'`: group commits by package.

>     -   `'aggregated'`: TODO:add description.

Reviewer: @Planeshifter Is this TODO intentional?

Author: Inserted.

-   **content**: Markdown formatted changelog.
-   **releaseType**: release type (`null` if changelog is for a non-release).

The function accepts the following `options`:

-   **releaseType**: a release type for which to generate the changelog.

-   **format**: changelog format. Must be one of the following:

    -   `'grouped'`: group commits by package.

>     -   `'aggregated'`: display all commits in a flat list without breaking changes up by package.

Reviewer: ```suggestion
    -   `'aggregated'`: display all commits in a flat list without grouping commits by package.
```

-   `auto`: automatically determine the release type based on parsed commit messages.
-   `none`: no release (equivalent to not specifying a release type).

> Under the hood, the changelog leverages `git log` to retrieve the commits from which to assemble the changelog. The `flags` option allows passing any options to the `git log` command, e.g. to generate a changelog for a specified time interval or for an individual contributor.

Reviewer: ```suggestion
When generating a changelog, the function uses `git log` to retrieve the commits from which to assemble a set of changes. The `flags` option allows passing options to directly to the `git log` command (e.g., to generate a changelog for a specified time interval or for an individual contributor).
```

> ```js

Reviewer: ```suggestion
```javascript
```

In stdlib, convention is to use `javascript`, not `js`.

var changelog = generate( '@stdlib/ndarray', {
    'flags': {
        'since': 'last month'
    }
});
// returns {...}

changelog = generate( '@stdlib/ndarray', {
    'flags': {
        'author': 'Athan Reines <kgryte@gmail.com>'
    }
});
// returns {...}
```

</section>

<!-- /.usage -->

<section class="notes">

> -   When `flags` are supplied, the generated changelog skips printing empty releases and uses a flat output format.

Reviewer: ```suggestion
-   When `flags` are supplied, the generated changelog does not include empty releases and uses a flat output format.
```

```javascript
// Changelog grouped by individual packages:
var changelog = generate( '@stdlib/math/base/utils', {
    'format': 'grouped'
});
// returns {...}

// Changelog where all commits, potentially touching many different packages, are merged together:
changelog = generate( '@stdlib/math/base/utils', {
    'format': 'aggregated'
});
// returns {...}
```

When generating a changelog, the function uses `git log` to retrieve the commits from which to assemble a set of changes. The `flags` option allows passing options to directly to the `git log` command (e.g., to generate a changelog for a specified time interval or for an individual contributor).

<!-- eslint-disable stdlib/doctest -->

```javascript
var changelog = generate( '@stdlib/ndarray', {
    'flags': {
        'since': 'last month'
    }
});
// returns {...}

changelog = generate( '@stdlib/ndarray', {
    'flags': {
        'author': 'Athan Reines <kgryte@gmail.com>'
    }
});
// returns {...}
```

</section>

<!-- /.usage -->

<section class="notes">

> -   When `flags` are supplied, the generated changelog does not include empty releases and uses a flat output format.

Reviewer: Is this note still accurate? It is not clear to me from the diff of `main.js` that this is true or that this constraint is necessary. I am mainly referring to the "flat" output format.

Reviewer: Also, it is not clear what the "flat" output format actually refers to.

Author: Good catch! Removed the note.
